pkg/controller/signup_test.go

```diff
@@ -212,9 +217,335 @@ func (s *TestSignupSuite) TestSignupGetHandler() {
+				return userSignup, nil
+			},
+		}
+
+		// Create a mock VerificationService
+		verifyService := &FakeVerificationService{
+			MockVerifyCode: func(ctx *gin.Context, signup *crtapi.UserSignup, code string) error {
+				return errors2.NewTooManyRequestsError("too many requests")
+			},
+		}
+
+		// Create Signup controller instance.
+		ctrl := controller.NewSignup(s.Config, svc, verifyService)
+		handler := gin.HandlerFunc(ctrl.VerifyCodeHandler)
+
+		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+		rr := httptest.NewRecorder()
+		ctx, _ := gin.CreateTestContext(rr)
+
+		req, err := http.NewRequest(http.MethodGet, "/api/v1/signup/verification/333333", nil)
+		require.NoError(s.T(), err)
+
+		ctx.Request = req
+		ctx.Set(context.SubKey, userID)
+		ctx.Params = append(ctx.Params, gin.Param{
+			Key:   "code",
+			Value: "333333",
+		})
+
+		handler(ctx)
+
+		// Check the status code is what we expect.
+		require.Equal(s.T(), http.StatusTooManyRequests, rr.Code)
+	})
+
+	s.Run("verifycode returns non status error", func() {
+		// Create a mock SignupService
+		svc := &FakeSignupService{
+			MockGetUserSignup: func(userID string) (userSignup *crtapi.UserSignup, e error) {
+				return &crtapi.UserSignup{
+					TypeMeta: v1.TypeMeta{},
+					ObjectMeta: v1.ObjectMeta{
+						Name: userID,
+						Annotations: map[string]string{
+							crtapi.UserSignupUserEmailAnnotationKey:        "jsmith@redhat.com",
+							crtapi.UserVerificationAttemptsAnnotationKey:   "0",
+							crtapi.UserSignupVerificationCodeAnnotationKey: "222222",
+							crtapi.UserVerificationExpiryAnnotationKey:     time.Now().Add(10 * time.Second).Format(verification.TimestampLayout),
+						},
+					},
+					Spec:   crtapi.UserSignupSpec{},
+					Status: crtapi.UserSignupStatus{},
+				}, nil
+			},
+			MockUpdateUserSignup: func(userSignup *crtapi.UserSignup) (userSignup2 *crtapi.UserSignup, e error) {
+				return userSignup, nil
+			},
+		}
+
+		// Create a mock VerificationService
+		verifyService := &FakeVerificationService{
+			MockVerifyCode: func(ctx *gin.Context, signup *crtapi.UserSignup, code string) error {
+				return errors.New("some other error")
+			},
+		}
+
+		// Create Signup controller instance.
+		ctrl := controller.NewSignup(s.Config, svc, verifyService)
+		handler := gin.HandlerFunc(ctrl.VerifyCodeHandler)
+
+		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+		rr := httptest.NewRecorder()
+		ctx, _ := gin.CreateTestContext(rr)
+
+		req, err := http.NewRequest(http.MethodGet, "/api/v1/signup/verification/222222", nil)
+		require.NoError(s.T(), err)
+
+		ctx.Request = req
+		ctx.Set(context.SubKey, userID)
+		ctx.Params = append(ctx.Params, gin.Param{
+			Key:   "code",
+			Value: "222222",
+		})
+
+		handler(ctx)
+
+		// Check the status code is what we expect.
+		require.Equal(s.T(), http.StatusInternalServerError, rr.Code)
```

Comment:
Do we just return a 500 error here and in the otter places or we do return some error details as a JSON so we could check the detailed error message too?

Just want to make sure we actually catch the expected (and different) errors and not the same 500 in all these tests.

Reply:
This would be dependent on what kind of error the verification service returns.   With the mock `VerifyCode` function we get the following body:

```
{"status":"Internal Server Error","code":500,"message":"some other error","details":"error while verifying code"}
```

Reply:
> This would be dependent on what kind of error the verification service returns. With the mock VerifyCode function we get the following body:

So, let's check that we do receive this error json.

Reply:
Done.